ironfish/src/rpc/routes/peers/getBannedPeers.ts

```diff
@@ -0,0 +1,80 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import * as yup from 'yup'
+import { PeerNetwork } from '../../../network'
+import { ApiNamespace, router } from '../router'
+
+export type BannedPeerResponse = {
+  identity: string
+  reason: string
+}
+
+export type GetBannedPeersRequest =
+  | undefined
+  | {
+      stream?: boolean
+    }
+
+export type GetBannedPeersResponse = {
+  peers: Array<BannedPeerResponse>
+}
+
+export const GetBannedPeersRequestSchema: yup.ObjectSchema<GetBannedPeersRequest> = yup
+  .object({
+    stream: yup.boolean().optional(),
+  })
+  .optional()
+  .default({})
+
+export const GetBannedPeersResponseSchema: yup.ObjectSchema<GetBannedPeersResponse> = yup
+  .object({
+    peers: yup
+      .array(
+        yup
+          .object({
+            identity: yup.string().defined(),
+            reason: yup.string().defined(),
+          })
+          .defined(),
+      )
+      .defined(),
+  })
+  .defined()
+
+router.register<typeof GetBannedPeersRequestSchema, GetBannedPeersResponse>(
+  `${ApiNamespace.peer}/getBannedPeers`,
+  GetBannedPeersRequestSchema,
+  (request, node): void => {
+    const peerNetwork = node.peerNetwork
+
+    if (!peerNetwork) {
+      request.end({ peers: [] })
+      return
+    }
```

Comment:
this is not nullable, you can delete this

```suggestion

```

Reply:
👍🏻